diff --git a/.cargo/config.toml b/.cargo/config.toml
index 10746d7..229495a 100644
--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -11,6 +11,13 @@ linker = "arm-linux-gnueabihf-gcc"
 linker = "aarch64-linux-musl-gcc"
 rustflags = ["-C", "target-feature=-crt-static"]
 
+[target.aarch64-unknown-linux-gnu]
+linker = "loongarch64-linux-gnu-gcc"
+
+[target.aarch64-unknown-linux-musl]
+linker = "loongarch64-linux-musl-gcc"
+rustflags = ["-C", "target-feature=+crt-static"]
+
 [target.wasm32-unknown-unknown]
 rustflags = [
     "-C",
diff --git a/scripts/build-npm.js b/scripts/build-npm.js
index ef447a7..ce12620 100644
--- a/scripts/build-npm.js
+++ b/scripts/build-npm.js
@@ -32,6 +32,14 @@ const triples = [
     name: 'aarch64-unknown-linux-musl',
     libc: 'musl',
   },
+  {
+    name: 'loongarch64-unknown-linux-gnu',
+    libc: 'glibc',
+  },
+  {
+    name: 'loongarch64-unknown-linux-musl',
+    libc: 'musl',
+  },
   {
     name: 'x86_64-unknown-linux-musl',
     libc: 'musl',
@@ -48,6 +56,7 @@ const cpuToNodeArch = {
   aarch64: 'arm64',
   i686: 'ia32',
   armv7: 'arm',
+  loongarch64: 'loong64',
 };
 const sysToNodePlatform = {
   linux: 'linux',
